src/robot_dart/robot.hpp

```diff
@@ -97,17 +98,17 @@ namespace robot_dart {
         Eigen::Vector6d com_velocity() const;
         Eigen::Vector6d com_acceleration() const;
 
-        Eigen::VectorXd positions() const;
-        void set_positions(const Eigen::VectorXd& positions);
+        Eigen::VectorXd positions(const std::vector<std::string>& dof_names = {}, bool filter_mimics = false, size_t start_dof = 0);
```

Comment:
Why do we need the `start_dof` parameter in those functions? Same applies to the `update()` method..

Also, why do we need the `filter_mimics` parameter?

Reply:
The start_dof was initially used in robot_control.cpp it is an offset on the command
The filter_mimics boolean activate the functionality which was here in robot_control.cpp : 
You can send a cmd vector which do not contain mimic vector zero commands, those will be automatically added

I don't know if we really want that, but because we want to remove the control part of robot_dart, I have added it here

Reply:
> The start_dof was initially used in robot_control.cpp it is an offset on the command

Yes, but start_dof was automatically identified in robot_control.

> You can send a cmd vector which do not contain mimic vector zero commands, those will be automatically added

But isn't `const std::vector<std::string>& dof_names` already doing that? You specify which dofs/joints to control with this. If you do not mention the mimic joints, they should receive zero.

> I don't know if we really want that, but because we want to remove the control part of robot_dart, I have added it here

I will handle this..

Reply:
> But isn't const std::vector<std::string>& dof_names already doing that? You specify which dofs/joints to control with this. If you do not mention the mimic joints, they should receive zero.

Yes exactly, I have just added the functions of robot_control.cpp because as mentioned [here](https://github.com/resibots/robot_dart/issues/42#issuecomment-628015685), it is not logical to handle dof order both in robot.cpp and robot_control.cpp

> I will handle this..

ok